utils/v2tofhirr4/mappings.bal

```diff
@@ -40,24 +40,27 @@ public function stringToHl7(string msg) returns hl7:Message|error {
 # + hl7 - HL7 message as a string or an hl7v2:Message  
 # + customMapper - Custom mapper implementation
 # + return - FHIR Bundle as a json
-public function v2ToFhir(string|hl7:Message hl7, SegmentToFhir? customMapper = ()) returns json|error {
+public isolated function v2ToFhir(string|hl7:Message hl7, SegmentToFhir? customMapper = ()) returns json|error {
     hl7:Message hl7msg;
     if (hl7 is string) {
         hl7msg = check stringToHl7(hl7);
     } else {
         hl7msg = hl7;
     }
-    SegmentToFhir mapperImpl = getMapperContext().getDefaultImpl();
-    if customMapper is () {
-        return check transformToFhir(hl7msg, mapperImpl);
+    SegmentToFhir mapperImpl;
+    lock {
+        mapperImpl = defaultMapperImpl.clone();
```

Comment:
shall we leave out `Impl` and use a name ends with `Mapper`?

Reply:
Fixed with 99c9318de23837d5a439bcb33a0d0a58a7035f8c